### ISSUE TITLE
Doesn't warn about unused variables

### ISSUE THREAD

**Issue by myknbani** [Microsoft/vscode-tslint#185]
I get the following reported from the terminal, but not inside vscode.  I get warnings for other things like missing semicolon or wrong alignment.

```
'x' is declared but never used. (no-unused-variable)
  21 | const setupHotReload = () => function plugin() {
  22 |     const app = this; // tslint:disable-line no-this
> 23 |     const x = 5;
```

```
All imports are unused. (no-unused-variable)
  10 | // + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +
  11 | 
> 12 | import { pipe, assoc, assocPath } from 'ramda';
```

**Comment by egamma** [comment]
This is a consequence of tslint 5.0 using the typechecker for the `no-unused-variable` rule (see the [release notes](https://github.com/palantir/tslint/releases/tag/5.0.0)) and vscode-tslint doesn't support rules which require the type checker (see #70). 

You have the following options:
- use tslint 4.5.1 (npm install tslint@4.5.1)
- enable the TypeScript provided check for unused variables in the `tsconfig.json`:
```json
"noUnusedLocals": true,
```

**Comment by myknbani** [comment]
Thanks!  I guess I'll just wait for #70 then.

**Comment by heinhoang** [comment]
I use angular 4 CLI, I choose to add `"noUnusedLocals": true` to `tsconfig.json` but why it's not working?

**Comment by jcrben** [comment]
maybe this will get smoother when https://github.com/Microsoft/vscode/issues/37212 is finished

**Comment by egamma** [comment]
@jcrben Microsoft/vscode#37212Microsoft/vscode#37212 is indeed a good example how to use the tsconfig.json compiler option `noUnusedLocals` instead of the tslint rule `no-unused-variables`. However, this work only applies to vscode code base itself, it is not a general solution.
